protocol/avdl/keybase1/teams.avdl

```diff
@@ -643,19 +644,41 @@ protocol teams {
   }
 
   record AnnotatedTeamInvite {
-    // metadata is taken straight from the protocol
+    // metadata is taken straight from TeamSigChainState
     TeamInviteMetadata inviteMetadata;
 
     TeamInviteDisplayName displayName;
     string inviterUsername;
-    UserVersion inviteeUv;
     string teamName;
 
-    // null for multiuse invites
-    union { null, TeamMemberStatus } status;
+    // isValid is whether an invite can still be completed.
+    // For single use invites, it means they are active and not completed or cancelled yet.
+    // For invitelinks, it means that they are active and (if an expiration time has been set) it
+    // has not expired, according to the client's local clock, and (if a max uses has been set) the
+    // maximum number of uses has not been reached.
+    boolean isValid;
+    // validityDescription is a description like "Expires after 12 days or 3 uses",
+    // "Expired after 12 uses", "Expired 100 years ago", "Does not expire".
+    string validityDescription;
```

Comment:
Hesitant about this change vs. creating this text in the gui. If the invite is expiring in on the order of hours you'd likely see a fairly outdated time here. What do you think?

Reply:
Right, that's why Y2K-1722 is going to send a notification on expiry or reload notifications if the link is expiring soon. Like reload every 30 minutes if its expiring in a day, every minute if its less than 2 hours, etc.

Reply:
for < 2 minutes we can either just do "Expiring soon" or reload every 3-4 seconds. I don't think the exact time is as important here compared to ephemeral messages

Reply:
I agree the it's not that important. The notification loop seems relatively tricky for something that might be solved by putting a timestamp field here, though.